test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -216,7 +216,7 @@ object Assertion {
   final def containsTheSameElements[A](other: Iterable[A]): Assertion[Iterable[A]] =
     Assertion.assertion("containsTheSameElements")(param(other)) { actual =>
       val actualSeq = actual.toSeq
-      val otherSeq = other.toSeq
+      val otherSeq  = other.toSeq
       actualSeq.diff(otherSeq).isEmpty && otherSeq.diff(actualSeq).isEmpty
```

Comment:
This definitely takes care of the assertion being accurate. What do you think about reporting? Right now if the assertion fails we are going to get something like:

```scala
List(1, 1, 2, 3) did not satisfy sameElements(List(1, 2, 3, 3))
```

In this case it is pretty obvious what the difference is but it could be harder to tell if the list is longer. I think if we made the two checks for the lists being empty into assertions themselves we could get something more like:

```scala
List(1, 1, 2, 3) did not satisfy sameElements(List(1, 2, 3, 3))
(List(1) did not satisfy isEmpty) && (List(3) did not satisfy isEmpty)
```

It isn't perfect because we're not exactly explaining to you what those sub-assertions are but we could put in some documentation about that. What do you think?

Reply:
+1 ... let's do as you suggest.